picard/util/tags.py

```diff
@@ -140,22 +258,93 @@ def insert(self, index, tagvar):
     def __repr__(self):
         return f"TagVars({self._items!r})"
 
-    def display_name(self, name):
-        tagdesc = None
+    def item_from_name(self, name):
         if ':' in name:
             name, tagdesc = name.split(':', 1)
+        else:
+            tagdesc = None
+
+        if name and name.startswith('_'):
+            search_name = name.replace('_', '~', 1)
+        elif name and name.startswith('~'):
+            search_name = name
+            name = name.replace('~', '_')
+        else:
+            search_name = name
+
+        item: TagVar = self._name2item.get(search_name, None)
+
+        return name, tagdesc, search_name, item
+
+    def script_name_from_name(self, name):
+        tagname, tagdesc, search_name, item = self.item_from_name(name)
+        if item:
+            return str(item)
+        return None
+
+    def display_name(self, name):
+        name, tagdesc, search_name, item = self.item_from_name(name)
 
-        item = self._name2item.get(name, None)
         if item and item.shortdesc:
             title = _(item.shortdesc)
         else:
-            title = name
-
+            title = search_name
         if tagdesc:
             return '%s [%s]' % (title, tagdesc)
         else:
             return title
 
+    @staticmethod
+    def _markdown(text: str):
+        if markdown is None:
+            return '<p>' + text.replace('\n', '<br />') + '</p>'
+        return markdown(text)
+
+    def _base_description(self, tagname):
+        name, tagdesc, _search_name, item = self.item_from_name(tagname)
+        content = self._markdown(_(item.longdesc) if item and item.longdesc else _(TEXT_NO_DESCRIPTION))
+        if tagdesc:
+            return f"<p><em>%{name}%</em> [{tagdesc}]</p>{content}"
+        else:
+            return f"<p><em>%{name}%</em></p>{content}"
```

Comment:
We should have formats in a translatable string (in case it varies with language).

```python
        fmt_tagdesc = _("<p><em>%{name}%</em> [{tagdesc}]</p>{content}")
        fmt_normal = _("<p><em>%{name}%</em></p>{content}")
        fmt = fmt_tagdesc if tagdesc else fmt_normal
        return fmt.format(name=name, content=content, tagdesc=tagdesc)
```

Reply:
Changed in https://github.com/metabrainz/picard/pull/2636/commits/48158eb412dd68c3b7906bedeef8e3be9bfb3ec1